_studio/mfx_lib/scheduler/src/mfx_scheduler_core.cpp

```diff
@@ -431,13 +427,13 @@ void mfxSchedulerCore::RegisterTaskDependencies(MFX_SCHEDULER_TASK  *pTask)
 
     // check if the table have empty position(s),
     // If so decrement the index of the last table entry.
-    i = m_numDependencies;
-    while ((0 < i) &&
-           (NULL == m_pDependencyTable[i - 1].p))
-    {
-        i -= 1;
-    }
-    m_numDependencies = i;
+    assert(m_pDependencyTable.size() > m_numDependencies);
+
+    i = (m_pDependencyTable.size() > m_numDependencies) ? m_numDependencies : 0;
```

Comment:
you writing an assert and then use the same condition as the correct one in following code... looks suspicious

Reply:
assert - only for debug.
what is wrong?
assert(true)// OK, so i = m_numDependencies
I have missed some thing?

Reply:
I mean that your code handles this condition (when assert is false) without any problems... So if the number of dependencies in array is incorrect, isn't it a fatal error?